backend/src/v2/component/launcher_v2.go

```diff
@@ -427,6 +433,46 @@ func prettyPrint(jsonStr string) string {
 
 const OutputMetadataFilepath = "/tmp/kfp_outputs/output_metadata.json"
 
+// We overwrite this as a DI mechanism for testing getLogWriter. We could make
+// it an explicit input to getLogWriter, but then it would have to be an input
+// in getLogWriter's caller as well.
+var osCreateFunc = os.Create
+
+// getLogWriter returns an io.Writer that can either be single-channel to stdout
+// or dual-channel to stdout AND a log file based on the URI of a log artifact
+// in the supplied ArtifactList. Downstream, the resulting log file gets
+// uploaded to the object store.
+func getLogWriter(artifacts map[string]*pipelinespec.ArtifactList) (writer io.Writer) {
+	logsArtifactList, ok := artifacts["logs"]
+
+	if !ok || len(logsArtifactList.Artifacts) == 0 {
+		return os.Stdout
+	}
+
+	logURI := logsArtifactList.Artifacts[0].Uri
+	logFilePath, err := LocalPathForURI(logURI)
+	if err != nil {
+		glog.Errorf("Error converting log artifact URI, %s, to file path.", logURI)
+		return os.Stdout
+	}
+
+	logFile, err := osCreateFunc(logFilePath)
+	if err != nil {
+		glog.Errorf("Error creating logFilePath, %s.", logFilePath)
+		return os.Stdout
+	}
+
+	return io.MultiWriter(os.Stdout, logFile)
+}
+
+// TODO: Clarify the differences between the various execution functions. Are
+// all of them necessary? Do the existing partitions make sense? Can the
+// function names more effectively encapsulate the division in responsibilities?
+// We have tried to clarify this with some  basic function doc strings, but a
+// deeper refactor might be advisable.
```

Comment:
while I agree with the spirit of this comment, I'm not sure I find this to be an appropriate place for this, a github issue would make more sense to me

Reply:
Agreed! Deleted.

Reply:
https://github.com/kubeflow/pipelines/issues/11791